auth-web/src/components/auth/Statements.vue

```diff
@@ -0,0 +1,200 @@
+  private readonly ITEMS_PER_PAGE = 5
+  private readonly PAGINATION_COUNTER_STEP = 4
+  private formatDate = CommonUtils.formatDisplayDate
+  private totalStatementsCount: number = 0
+  private tableDataOptions: any = {}
+  private isDataLoading: boolean = false
+  private statementsList: StatementListItem[] = []
+
+  private readonly headerStatements = [
+    {
+      text: 'Date',
+      align: 'left',
+      sortable: false,
+      value: 'dateRange',
+      width: '270'
+    },
+    {
+      text: 'Frequency',
+      align: 'left',
+      sortable: false,
+      value: 'frequency',
+      width: '140'
+    },
+    {
+      text: 'Downloads',
+      align: 'right',
+      sortable: false,
+      value: 'action'
+    }
+  ]
+
+  private get getPaginationOptions () {
+    return [...Array(this.PAGINATION_COUNTER_STEP)].map((value, index) => this.ITEMS_PER_PAGE * (index + 1))
+  }
+
+  private async mounted () {
+    this.setAccountChangedHandler(this.initialize)
+    this.initialize()
+  }
+
+  private async initialize () {
+    await this.loadStatementsList(1, this.ITEMS_PER_PAGE * 2)
+  }
+
+  formatDateRange (date1, date2) {
+    let displayDate = ''
+    let dateObj1 = this.getMomentDateObj(date1)
+    let dateObj2 = this.getMomentDateObj(date2)
+    let year = (dateObj1.year() === dateObj2.year()) ? dateObj1.year() : ''
+    let month = (dateObj1.month() === dateObj2.month()) ? dateObj1.format('MMMM') : ''
+    if (date1 === date2) {
+      return dateObj1.format('MMMM DD, YYYY')
+    } else if (year && !month) {
+      return `${dateObj1.format('MMMM DD')} - ${dateObj2.format('MMMM DD')}, ${year}`
+    } else if (year && month) {
+      return `${month} ${dateObj1.date()} - ${dateObj2.date()}, ${year}`
+    } else {
+      return `${dateObj1.format('MMMM DD, YYYY')} - ${dateObj2.format('MMMM DD, YYYY')}`
+    }
+  }
+
+  getMomentDateObj (dateStr) {
+    return moment(dateStr, 'YYYY-MM-DD')
+  }
+
+  @Watch('tableDataOptions', { deep: true })
+  async getTransactions (val, oldVal) {
+    const pageNumber = val.page || 1
+    const itemsPerPage = val.itemsPerPage
+    await this.loadStatementsList(pageNumber, itemsPerPage)
+  }
+
+  private async loadStatementsList (pageNumber?: number, itemsPerPage?: number) {
+    this.isDataLoading = true
+    const filterParams: StatementFilterParams = {
+      pageNumber: pageNumber,
+      pageLimit: itemsPerPage
+    }
+    const resp = await this.getStatementsList(filterParams)
+    this.statementsList = resp?.items || []
+    this.totalStatementsCount = resp?.total || 0
+    this.isDataLoading = false
+  }
+
+  private get isStatementsAllowed (): boolean {
+    return (this.currentOrganization?.orgType === Account.PREMIUM) &&
+      [MembershipType.Admin, MembershipType.Coordinator].includes(this.currentMembership.membershipTypeCode)
+  }
+
+  private customSortActive (items, index, isDescending) {
```

Comment:
i have seen this code in many places...anyway to make it generic?

Reply:
we can place it in the common utils